addons/website_slides/controllers/main.py

```diff
@@ -524,6 +648,32 @@ def slide_category_search_read(self, fields, domain):
             'can_create': can_create,
         }
 
+    @http.route('/slides/add_category', type="json", website=True, auth="user")
+    def add_category(self, channel_id, name, **kw):
+        channel = request.env['slide.channel'].browse(channel_id)
+        request.env['slide.category'].create({
+            'name': name,
+            'channel_id': channel.id
+        })
+
+        return {'url': "/slides/%s" %(slug(channel))}
+
+    #Not using the /web/dataset/resequence route as a slide can be dragged into another category at the same time
+    @http.route('/slides/resequence_slides', type="json", website=True, auth="user")
+    def resequence_slides(self, slides_data=None, **kw):
+        channel_user = None
+        for slide in slides_data:
+            s = request.env['slide.slide'].browse(slide['id'])
```

Comment:
Should collect all ids and do a single search to increase performances.
Something like `request.env['slide.slide'].search([('id', 'in', [slide['id'] for slide in slides_data])])`

And then loop on result.

Reply:
aaand name it slide, instead of s, because it's more self_speaking :)
Like : 
```
for slide in request.env['slide.slide'].search([('id', 'in', [slide['id'] for slide in slides_data])]):
   doStuff()
```